Library/Homebrew/tap.rb

```diff
@@ -1169,7 +1169,8 @@ def formula_files_by_name
       name, formula_hash = item
       # If there's more than one item with the same path: use the longer one to prioritise more specific results.
       existing_path = hash[name]
-      new_path = File.join(tap_path, formula_hash["ruby_source_path"]) # Pathname equivalent is slow in a tight loop
+      # Pathname equivalent is slow in a tight loop
+      new_path = File.join(tap_path, formula_hash.fetch("ruby_source_path"))
```

Comment:
`File#join` expects one or more strings while `Array#join` handles nil arguments gracefully.

```rb
irb(main):001:0> ["one", nil].join
=> "one"
irb(main):002:0> File.join("one", nil)
(irb):2:in `join': no implicit conversion of nil into String (TypeError)
        from (irb):2:in `<main>'                              
        from /Users/kevinrobell/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/irb-1.6.2/exe/irb:11:in `<top (required)>'
        from /Users/kevinrobell/.asdf/installs/ruby/3.2.2/bin/irb:25:in `load'
        from /Users/kevinrobell/.asdf/installs/ruby/3.2.2/bin/irb:25:in `<main>'
```

Reply:
As long as the API JSON is current, `"ruby_source_path"` will always be populated, right?

Reply:
> As long as the API JSON is current, `"ruby_source_path"` will always be populated, right?

Correct.

> In any case this doesn't change the behaviour in case of `nil`, only the error message.

Yep, and it also better shows intent.